addon/-private/system/relationships/state/relationship.js

```diff
@@ -266,6 +269,42 @@ export default class Relationship {
     this.flushCanonicalLater();
   }
 
+  /*
+    Call this method once a record deletion has been persisted
+    to purge it from both current and canonical state of all
+    relationships.
+
+    @method removeDeletedInternalModelFromInverse
+    @private
+   */
+  removeDeletedInternalModelFromInverse() {
```

Comment:
I believe `Deleted` should be removed from this record name. In general a method name should describe `what` it does, not why or when it should be invoked.